src/components/Form.js

```diff
@@ -1,27 +1,68 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { addBook } from '../redux/books/books';
 import './styles/form.css';
 
 const Form = () => {
+  const dispatch = useDispatch();
+
+  const bookStore = () => ({
+    id: '',
+    title: '',
+    author: '',
+    category: '',
+  });
+
+  const [bookSetup, setBookSetup] = useState('');
```

Comment:
- [ ] It will be good to set up the default state to an object with values `{  title: '', author: '', category: '' }` instead of an empty string `" "`. This is because you are using an object to represent your form data and to make your form controlled.  Also, since you have not predefined your values in the object state, react is complaining. We are getting the error below when you try to fill in the values in your form fields. Kindly check the console to see the error.

**error we are getting**

`react-dom.development.js:67 Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://reactjs.org/link/controlled-components`